math/minuit2/inc/Minuit2/ExternalInternalGradientCalculator.h

```diff
@@ -0,0 +1,42 @@
+// @(#)root/minuit2:$Id$
+// Authors: M. Winkler, F. James, L. Moneta, A. Zsenei, E.G.P. Bos   2003-2017
+
+/**********************************************************************
+ *                                                                    *
+ * Copyright (c) 2005 LCG ROOT Math team,  CERN/PH-SFT                *
+ * Copyright (c) 2017 Patrick Bos, Netherlands eScience Center        *
```

Comment:
Do you really want to keep this class name instead of calling something else ?

Reply:
I'm open to suggestions, I don't love it either.

Reply:
Maybe `InternalSpaceGradientCalculator`? Or `MinuitSpaceGradientCalculator`? Or `TransformedSpaceGradientCalculator`? I find it hard to make it more compact, while at the same time clearly indicating the relation to the other gradient calculators.

Perhaps as an alternative it would make sense to make a namespace `GradientCalculator` in which the different types could be gathered.

Reply:
To at least make the use case clearer (and perhaps also the name), I added some doxygen documentation for the class. Hope this helps.